### DISCUSSION TITLE
Any HA's gurus here?

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#104] [General]
When comes to HA i still have way too much questions so I thought of trying asking you guys here :-)

1. `state_class` of a sensor. It's clear that for Total* sensors it's `total_increasing` but what about Today's* sensors? Shouldn't it just be a `total`?

**Comment by gedger** [comment]
From my understanding then yes the today sensor can be total_increasing. The best reference I have found is.

https://developers.home-assistant.io/blog/2021/08/16/state_class_total/

> For sensors with state_class STATE_CLASS_TOTAL_INCREASING, a decreasing value is interpreted as the start of a new meter cycle or the replacement of the meter. It is important that the integration ensures that the value cannot erroneously decrease in the case of calculating a value from a sensor with measurement noise present. This state class is useful for gas meters, electricity meters, water meters etc.

**Reply by davidrapan** [reply]
Why it does not work for template sensors is beyond me then.

_Always complains about decreased value..._

**Reply by gedger** [reply]
Have a detailed look at the raw data for the sensor. I've had this and it turned out to be the sensor data having very small downward glitches. The total_increasing also needs an obvious change to signify a start of new cycle. I have some heaters in my workshop which aren't used this time of year but the sensors still report small occasional power use of 1W (noise) and when these are reset to zero the change again triggers the total_increasing error.

**Reply by davidrapan** [reply]
Ehm it has total_increasing with power device_class? Or did I misunderstood something?

**Reply by gedger** [reply]
Yup power device_class and total_increasing state_class but the data was invalid when I looked at it in detail. I've also had issues with how Python stores numbers where 0 is appears to be stored as -0.0000000123 which again triggers an error.
